Modules/Core/Common/include/itkBoundingBox.h

```diff
@@ -90,7 +90,7 @@ class ITK_TEMPLATE_EXPORT BoundingBox : public Object
   /** Hold on to the type information specified by the template parameters. */
   using PointIdentifier = TPointIdentifier;
   using CoordinateType = TCoordinate;
-#ifndef ITK_FUTURE_LEGACY_REMOVE
+#ifndef ITK_REMOVE_COORD_REP_TYPES
```

Comment:
Can an advanced option be created for turning this on?
```cmake
# The ITK_REMOVE_COORD_REP_TYPES option is a transient flag to assist with API changes
# It is a flag expected to be removed after ITK version 6.0 release.  It is not intended to be part of the public API.
option(ITK_REMOVE_COORD_REP_TYPES OFF)
mark_as_advanced(ITK_REMOVE_COORD_REP_TYPES)
'''

Reply:
>  Because when users need to support _both_ ITK 6.0 and ITK 5.4 in one and the same project, I would rather not see them "littering" their code with `#if ITK_VERSION_MAJOR < 6 CoordRepType #else CoordinateType `.

I think the mistake is trying to support 5.4 and 6.0 with FUTURE_LEGACY_REMOVE. You should be able to compile ITK 6.0 with LEGACY_REMOVE, and be much closer to support 5.4 too.

Reply:
@blowekamp Thanks, but then if you compile user code with ITK 6.0 + `LEGACY_REMOVE=ON`, you might get warnings, when you still use those deprecated `CoordRepType` aliases in your user code. Is that OK? (Again, I'd rather not see users writing `#if ITK_VERSION_MAJOR < 6 CoordRepType #else CoordinateType` in their code.)

Reply:
Can you compile with LEGACY_SILENT until support for 5 is dropped?

Reply:
The `CoordRepType` are currently marked by `ITK_FUTURE_DEPRECATED(message)`, which evaluates to `[[deprecated(message)]]` when `LEGACY_REMOVE` is ON:

https://github.com/InsightSoftwareConsortium/ITK/blob/6a704aaadef717113aa80e7e021c408c086b3677/Modules/Core/Common/include/itkMacro.h#L1378-L1382

Reply:
I'm still 100% behind renaming CoordRepType to CoordinateType, I just hope that it doesn't annoy anyone  😇 !

Reply:
> Another option would be to back port the new typedef to release-5.4 branch.

Thanks @blowekamp  I think it would indeed be nice if we could cherry-pick these two for the release-5.4 branch:
- fbe9b046b21d9a1353be13512344cdc839fd15b4
- 5ea1a9a8dfc322cc4ae4c18b09136d27c554b31b

These two would just make the new CoordinateType aliases available to ITK 5.4 users, without any deprecate or removal. What do you think?

Reply:
- Please have a look at PR #5014